### PR TITLE
Use SQLite write-ahead logging and shared caches

### PR DESCRIPTION
**Pull request by bricelam**
This will result in less locking and, thus, greater throughput.

cc @ajcvickers @sebastienros

### CODE REVIEW

File: src/ProjectTemplates/Web.Spa.ProjectTemplates/content/Angular-CSharp/appsettings.json [hunk 1]
```diff
@@ -4,15 +4,15 @@
...
 //  },
 ////#endif
   "Logging": {
       "LogLevel": {
-      "Default": "Information",
-      "Microsoft": "Warning",
-      "Microsoft.Hosting.Lifetime": "Information"
+      "Default": "Information",
+      "Microsoft": "Warning",
+      "Microsoft.Hosting.Lifetime": "Information"
```
Comment (bricelam) [inline src/ProjectTemplates/Web.Spa.ProjectTemplates/content/Angular-CSharp/appsettings.json:15]: ⁉ Did this file have mixed Unix and Windows line endings?

**Comment by bricelam** [comment]
> Binary file not shown.

I ran `PRAGMA journal_mode = 'wal';` on these files.

**Comment by bricelam** [comment]
SQLite docs: (for the curious)

* [Write-Ahead Logging](https://www.sqlite.org/wal.html)
* [Shared-Cache Mode](https://sqlite.org/sharedcache.html)

**Comment by ajcvickers** [comment]
@Pilchie This is a change to the SQLite connection strings that we give to customers in various places in the ASP.NET Core code. Who would be an appropriate person to review this?

**Comment by Pilchie** [comment]
@mkArtakMSFT - this mostly affects templates I think.  Can you find someone to take a look?

File: src/ProjectTemplates/Web.Spa.ProjectTemplates/content/Angular-CSharp/appsettings.json (hunk 1 above)
Comment (mkArtakMSFT) [reply src/ProjectTemplates/Web.Spa.ProjectTemplates/content/Angular-CSharp/appsettings.json:15]: Definitely not intentional, if that was the case.

**Review by mkArtakMSFT — APPROVED** [review]
Thanks @bricelam!

**Comment by bricelam** [comment]
Should I make similar changes to the blazor-wasm branch?

**Review by captainsafia — COMMENTED** [review]
Is there a reason that the database files are included in this commit?

Looks sensible otherwise.

**Comment by mkArtakMSFT** [comment]
> 
> 
> Should I make similar changes to the blazor-wasm branch?

Thanks @bricelam. We're locking down for the Blazor WASM release already and limit the fixes we take. We won't take this there. In about a month we will start merging blazor-wasm into release/3.1.

**Comment by bricelam** [comment]
@captainsafia they’re checked in for the templates experience (so you don’t have to run `dotnet ef database update` with SQLite). I just updated them to enable write-ahead logging.

**Comment by bricelam** [comment]
@sebastienros Do we know if this had any impact on throughput? Is there a report showing the BasicApi or BasicViews tests against SQLite?

**Pull request merged**